code/__HELPERS/dynamic_human_icon_gen.dm

```diff
@@ -0,0 +1,55 @@
+GLOBAL_LIST_EMPTY(dynamic_human_icons)
+
+/proc/get_dynamic_human_icon(outfit_path, species_path = /datum/species/human, mob_spawn_path, r_hand, l_hand, bloody_slots = NONE, generated_dirs = GLOB.cardinals)
+	if(!species_path)
+		return FALSE
+	var/arg_string = "[outfit_path]_[species_path]_[mob_spawn_path]_[l_hand]_[r_hand]_[bloody_slots]_[english_list(generated_dirs, nothing_text = "", and_text = ",", comma_text = ",")]"
+	if(GLOB.dynamic_human_icons[arg_string])
+		return GLOB.dynamic_human_icons[arg_string]
+	var/mob/living/carbon/human/dummy/consistent/dummy = new()
+	dummy.set_species(species_path)
+	dummy.stat = DEAD //this is to avoid side effects of mob spawners
```

Comment:
mob spawners?

Reply:
mob spawners that we use for icon effects, it calls death(), and having stat = dead means you early return on death